pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -170,6 +170,18 @@ func (i *Instance) TryRemoveFinalizer() bool {
 	return false
 }
 
+func (p *Parameter) IsImmutable() bool {
+	return p.Immutable != nil && *p.Immutable
+}
+
+func (p *Parameter) IsRequired() bool {
+	return p.Required != nil && *p.Required
+}
+
+func (p *Parameter) HasDefault() bool {
+	return p.Default != nil
```

Comment:
The nil check is here, right? Did you mean a check against empty string?

I thought about it but decided against it, because an empty string might be a default? I'm not fully sure how we handle that though.